### ISSUE TITLE
Automatic grouping of bookmarks per country

### ISSUE THREAD

**Issue by GutHib** [organicmaps/organicmaps#5290]
Hi all,

In OM I always (try to?) organize my bookmarks in country lists, and I'm assuming it's what most people do who travel a lot. That's the theory, at least, because in practice, I always end up with a huge mess and bookmarks all over the place. :/

To me it would be great if OM could do the job automatically and add bookmarks to country lists - either upon creation or maybe even afterwards ("take bookmarks from given list and organize them per country").

Thanks for considering!

**Comment by GutHib** [comment]
Hi - is there any chance that this will be added eventually, or does the fact that it hasn't received any replies and dropped to page 42 on the issues list mean it is dead in the water? Thank you!

**Comment by Jean-BaptisteC** [comment]
This feature can be implemented in the future, if we have enough time, don't forget we work on OM on our free time. Make donation can give the ability to develop more features.

**Comment by GutHib** [comment]
I already donated for this project two years ago, and it didn't lead to any of my suggestions being implemented (not that I expected that).

Hence I make a suggestion first. If it takes off, fine, then I might send a few beers or a cup of coffee as a thank you. In this case, my follow-up wasn't so much addressed at the developers - rather I was wondering why there are no other users picking up on it.

PS: So the question about life, the universe and everything else is: "Which page did my Organic Maps suggestion drop down to?" Who would have thought...

**Comment by biodranik** [comment]
@GutHib don't worry, we monitor _every_ issue, it is not lost. This one is very convenient, so we will implement it when we have more time/resources, and other critical issues will be fixed. Or when any contributor decides to do it.

Sorting bookmarks by location should be done carefully, to get fast results in cases with thousands of bookmarks. It would be great if anyone could share large bookmark datasets from different cases to test this feature.

**Comment by GutHib** [comment]
That is great to know. I can provide a set of around 1800 bookmarks from around the world, including a few tracks. I reckon you will treat them with privacy and use them for testing only - I would hate to run into a few thousand OM users next time I visit my favorite mushroom grounds. ;)

**Comment by pastk** [comment]
I'm not sure automatic creation of per-country lists is a good idea, as different people might use different categorization for their bookmarks.. It could be a nice optional feature though.

Alternatively, we might make bookmarks more easily searchable and/or sortable by country/location?

**Comment by GutHib** [comment]
Well, I'm a folder guy - I also prefer storing mp3s in physical folders rather than having them all over the place and let the mp3 player do the sorting. 

I guess what I'm saying is there are different approaches - but OM seems to be folder-based as well, otherwise it wouldn't make sense to have bookmark folders in the first place.

And if you do have folders, it's annoying to have your bookmarks end up in the wrong place (sorting bookmarks manually can be tedious). So more broadly, the idea would be to give some assistance and make misplacement of bookmarks less likely. :)

- In my case, it's clearly countries (or regions).
- Sombody else might prefer to have their bookmarks separated by colors.
- Not sure if OM can recognize the type of places you have bookmarked (e.g. bar vs. pub, or gallery vs. monument) - if yes, that would be an option as well.

Of course that would make the whole thing more complicated to implement.

**Comment by pastk** [comment]
> 
> And if you do have folders, it's annoying to have your bookmarks end up in the wrong place (sorting bookmarks manually can be tedious). So more broadly, the idea would be to give some assistance and make misplacement of bookmarks less likely. :)
> 

True, my bookmarks often end up in a wrong list just because <strike>I'm too lazy</strike> its too tedious to change list for a new bookmark
- https://github.com/organicmaps/organicmaps/issues/6234

**Comment by GutHib** [comment]
> > True, my bookmarks often end up in a wrong list just because I'm too lazy its too tedious to change list for a new bookmark

What kind of lists are those - not country-lists, I suppose?

**Comment by pastk** [comment]
I have a mixture - some (legacy) country-based lists and (lately) topic based ones.
I used to categorize by country before but in last years I avoid to do so as I realized there is no value in such categorization for me.
